Topics/User_Experience/Fuzzy_Search.md

```diff
@@ -0,0 +1,52 @@
+  - Typos are inevitable! This is true both when users are querying some collection of data, and also when they are adding new entries for others to search (e.g. a seller mistakenly listing a new `pencil caes` on Amazon). The more accommodating result selection afforded by fuzzy search reduces the likelihood that users are unable to find their desired result and draw incorrect conclusions from it.
+- [6: Low Physical Effort](https://universaldesign.ie/what-is-universal-design/the-7-principles/#p6)
+  - For users that may be hazy on what they are searching for, an exact-match algorithm will likely require multiple, repetitive searches from them to find their target result. This can not only lead to fatigue while using the application, but also exacerbate accessibility issues for users who cannot input text as comfortably.
+ 
+Given that the majority of applications have some use case for a search bar, **being able to implement a user-friendly search experience is essential in many software development projects**. Careful consideration should be given to any search tool you implement regardless of whether you choose fuzzy search or a more sophisticated algorithm. Moreover, the core idea behind fuzzy search - approximate string matching - is highly useful in designing many non-search tools a developer may be interested in, like autocorrect or data processors.
+ 
+## Implementation: Levenshtein distance
+In this section, we discuss the [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance) and how to use it in implementing fuzzy search.
+
+Levenshtein is a popular distance metric to compare the similarity between a query string and a target string. Intuitively, it counts the fewest number of times a character in the query must be deleted, inserted, or replaced to reach the target. For instance, if the query is `Brian`, its Levenshtein distance to `Bryan` is 1 as it only needs to replace `i` with `y` to reach the target. Notice that this can also be done in two operations by deleting `i` and inserting `y`, but this is suboptimal. Targets with a lower Levenshtein score are considered closer to the query.
+
+The algorithm for computing Levenshtein distance is not overly complicated and can be a fun exercise in designing recursive functions. [This GeeksforGeeks article](https://www.geeksforgeeks.org/introduction-to-levenshtein-distance/) describes several possible implementations in different programming languages should you decide to code it yourself.
+
+That being said, you can also find out-of-the-box solutions online that are likely better optimized. For applications built in JavaScript, the [js-levenshtein](https://www.npmjs.com/package/js-levenshtein) library is quite efficient and easy to use:
+- Install it by running `npm install --save js-levenshtein`.
+- In your code, import the module using `const levenshtein = require('js-levenshtein');`
```

Comment:
Since most projects use js modules, could it be worth specifying to use either require('js-levenshtein') or import {x} from 'js-levenshtein'

Reply:
Great catch, I'll make that change as well.